newIDE/app/public/external/piskel/piskel-main.js

```diff
@@ -281,87 +153,204 @@ function piskelCreateAnimation(pskl, piskelOptions) {
+                  );
+
+                  piskelController.addFrameAtCurrentIndex();
+                  piskelController.selectNextFrame();
+                  const currentFrameObj = piskelController.getCurrentFrame();
+                  pskl.utils.FrameUtils.addImageToFrame(
+                    currentFrameObj,
+                    image,
+                    0,
+                    0
+                  );
+                  pskl.tools.transform.TransformUtils.center(currentFrameObj);
+
+                  piskelController.getLayers().forEach(layer => {
+                    layer.moveFrame(
+                      piskelController.getCurrentFrameIndex(),
+                      frameIndex
+                    );
+                    layer.getFrameAt(frameIndex).originalIndex = frameIndex;
+                    layer.getFrameAt(
+                      frameIndex
+                    ).originalPath = flattenedFramePath;
+                  });
+                });
+              }
+            );
+          }
+        });
+
+        // Remove any frames that were removed in GD
+        const layer = piskelController.getLayerAt(0);
+        editorDataPaths.forEach((metaFramePath, index) => {
+          if (!flattenedImagePaths.includes(metaFramePath)) {
+            for (let fi = 0; fi < piskelController.getFrameCount(); fi++) {
+              if (metaFramePath === layer.getFrameAt(fi).originalPath) {
+                piskelController.getLayers().forEach(layer => {
+                  layer.removeFrameAt(fi);
+                });
+              }
+            }
+          }
+        });
+
+        // Put frames in the same order as they are in GD
+        piskelController.getLayers().forEach(layer => {
+          layer.getFrames().sort((a, b) => {
+            return (
+              flattenedImagePaths.indexOf(a.originalPath) -
+              flattenedImagePaths.indexOf(b.originalPath)
+            );
+          });
+        });
       }
-    },
-    function(err) {
-      // Finally load the image objects into piskel
-      const piskelFile = pskl.service.ImportService.prototype.createPiskelFromImages_(
-        imageData,
-        piskelOptions.name,
-        maxWidth,
-        maxHeight,
-        false
-      );
-      pskl.app.piskelController.setPiskel(piskelFile, {});
-      pskl.app.piskelController.setFPS(piskelOptions.fps);
-
-      // Add original path variable to imported frame objects, so we can overwrite them later when saving changes
-      const layer = pskl.app.piskelController.getCurrentLayer();
-      for (let i = 0; i < pskl.app.piskelController.getFrameCount(); i++) {
-        layer.getFrameAt(i).originalPath =
-          piskelOptions.resources[i].resourcePath;
-        layer.getFrameAt(i).originalName =
-          piskelOptions.resources[i].resourceName;
-        layer.getFrameAt(i).originalIndex = i;
+    );
+  } else {
+    // Load flat images into piskel if there is no metadata - the old way
```

Comment:
What do you mean by the "old way" here? Is that deprecated? In this case you should write "deprecated" rather than old. Otherwise, if it's not deprecated, don't write "old" at all because this notion of time does not exist in the source code :)

Reply:
It's still in use - so not deprecated. I will change the description to avoid confusion

Reply:
Cool thanks :)